openrarity/metrics/trait_count.py

```diff
@@ -0,0 +1,50 @@
+from openrarity.token.types import AttributeStatistic, MetadataAttribute
+
+
+def unique_trait_count(tokens: list[AttributeStatistic]) -> list[AttributeStatistic]:
+    """Add a statistic to each token with the count of unique traits on that token.
+
+    Parameters
+    ----------
+    tokens : list[AttributeStatistic]
+        Tokens to be augmented.
+
+    Returns
+    -------
+    list[AttributeStatistic]
+        Augmented Tokens
+    """
+    return []
+
+
+def count_traits(
+    attributes: list[MetadataAttribute],
+) -> list[MetadataAttribute]:
+    """Count the number of traits on a given token and append it as an attribute named
```

Comment:
not sure i understand the logic , how it's working ? in the code is can see how we append something to an array.

Reply:
The code uses list comprehension to generate the array with the token and its count. Documentation can be better though.

In python doing
```python
lst = []
for val in something:
  lst.append(val)
```
is significantly less performant than list comprehensions. List comprehensions are very pythonic so I am going to fight back against removing them.